internal/manager/config.go

```diff
@@ -64,17 +62,17 @@ type Config struct {
 	UpdateStatus         bool
 
 	// Kubernetes API toggling
-	IngressExtV1beta1Enabled util.EnablementStatus
-	IngressNetV1beta1Enabled util.EnablementStatus
-	IngressNetV1Enabled      util.EnablementStatus
-	UDPIngressEnabled        util.EnablementStatus
-	TCPIngressEnabled        util.EnablementStatus
-	KongIngressEnabled       util.EnablementStatus
-	KnativeIngressEnabled    util.EnablementStatus
-	KongClusterPluginEnabled util.EnablementStatus
-	KongPluginEnabled        util.EnablementStatus
-	KongConsumerEnabled      util.EnablementStatus
-	ServiceEnabled           util.EnablementStatus
+	IngressExtV1beta1Disabled bool
```

Comment:
dont you think the EnablementStatus works as a bool coz it mainly have Enable/Disable, though #1585 introduce the auto concept ? Basically, we need a flag toggle controller enablement during start time, which is one-time thing.

Reply:
Discussion in #1580 sums up why I think it makes more sense not to distinguish auto and enabled. tl;dr booleans are easier to work with and there's no realistic scenario where an end user would actually need to distinguish between auto and enabled. The auto-handlers aren't something users control directly, they're just internal logic we need to safely handle enabled controllers in some environments.